### PR TITLE
#14-Regarding the Back End-DelroyGayle

### PR DESCRIPTION
**Pull request by DelroyGayle**
## Description
Created the schemas and mock data for 5 databases

1. User
2. Initiative
3. Programme
4. SchoolStats
5. SchoolSurveys

## Related to
#14

Reviewers please check that I have not missed a field

I could not remember the different types of users that we decided so I went with

Super Admin, Country Manager and General User

Of course, everything is subject to change so please point out the correct titles.
Thank you

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] I have updated any documentation

### CODE REVIEW

File: server/db/initiatives-migration.sql [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+-- Create Initiative Table
+CREATE TABLE initiative (
+   id serial PRIMARY KEY,
+   name VARCHAR(50) UNIQUE NOT NULL,
+   category VARCHAR(50) NOT NULL,
+   description VARCHAR(255) NOT NULL,
+   createdById INT NOT NULL,
+   createdAt TIMESTAMP NOT NULL DEFAULT NOW(),
+   updatedById INT NOT NULL,
+   updatedAt TIMESTAMP NOT NULL DEFAULT NOW()
```
Comment (berkeli) [inline server/db/initiatives-migration.sql:11]: I don't think we need updated by, if we are adding updated by then we need to add the relevant reference
```suggestion
   updatedById INT NOT NULL,
   updatedAt TIMESTAMP NOT NULL DEFAULT NOW()
   FOREIGN KEY (updatedById) REFERENCES users (id)
   FOREIGN KEY (createdById) REFERENCES users (id)
```

File: server/db/programme-migration.sql [hunk 2]
```diff
@@ -0,0 +1,135 @@
+DROP TABLE IF EXISTS Programme;
+-- Create Programme Table
+CREATE TABLE programme (
```
Comment (berkeli) [inline server/db/programme-migration.sql:3]: programme and initiative are both the same thing

File: server/db/user-migration.sql [hunk 3]
```diff
@@ -0,0 +1,135 @@
+DROP TABLE IF EXISTS users;
+-- Create Users Table
+CREATE TABLE users (
```
Comment (berkeli) [inline server/db/user-migration.sql:3]: we should follow same naming scheme, i.e. not plural (programme, initiative, user)
```suggestion
CREATE TABLE user (
```

File: server/db/user-migration.sql [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+   id serial PRIMARY KEY,
+   userName VARCHAR(50) NOT NULL,
```
Comment (berkeli) [inline server/db/user-migration.sql:5]: We will not be using usernames, only email + password for login

File: server/db/user-migration.sql [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+   email VARCHAR(50) UNIQUE NOT NULL,
+   role VARCHAR(50) NOT NULL,
+   createdById INT NOT NULL,
+   createdAt TIMESTAMP NOT NULL DEFAULT NOW(),
+   updatedById INT NOT NULL,
+   updatedAt TIMESTAMP NOT NULL DEFAULT NOW()
```
Comment (berkeli) [inline server/db/user-migration.sql:11]: I don't think we need this information on user
instead we can add the country of the user
```suggestion
 country VARCHAR(50) NOT NULL
```

File: server/routes/api.js [hunk 6]
```diff
@@ -17,6 +27,44 @@ router.get("/", (_, res) => {
...
+			let reply = result.rows;
+			if (reply.length === 0) {
+				genericIntErrorMessage(userId, response, 2, "User");
+				return;
+			}
+
+			return response.json(reply); // Success
+		}
+	});
+});
```
Comment (berkeli) [inline server/routes/api.js:66]: Please note the users endpoint has already been implemented and we get users from auth0. I'm not sure why you are working on users endpoints

**Review by berkeli — CHANGES_REQUESTED** [review]
Thanks @DelroyGayle Looks good, but please remove users endpoints as they are already implemented and your code is overriding what I have implemented for the users endpoint.

**Comment by berkeli** [comment]
Also, there's one more table missing - ScheduleTracker
It should have the following fields

Programme/Initiative  -> reference to programme
School/NGO name -> reference to school
No. of new students reached - number
No. of existing students reached - number
No. of new teachers reached- number
No. of existing students reached - number
No. of tablets (if applicable)- number
Grades - text (for now, at least 255 chars )
Languages Taught - text (for now, at least 255 chars )
createdById -> -> reference to user

no need to create mock data for this

**Comment by DelroyGayle** [comment]
> please remove users endpoints as they are already implemented and your code is overriding what I have implemented for the users endpoint.

**Pull request closed without merging**